packages/grid/_modules_/grid/hooks/features/rows/gridRowsState.ts

```diff
@@ -1,4 +1,28 @@
-import { GridRowGroupingResult } from '../../core/rowGroupsPerProcessing';
+import type { GridRowGroupingResult, GridRowGroupParams } from '../../core/rowGroupsPerProcessing';
+import { DataGridProcessedProps } from '../../../models/props/DataGridProps';
+import { GridRowsProp } from '../../../models/gridRows';
+
+export type GridRowInternalCacheValue = Omit<GridRowGroupParams, 'previousTree'>;
+
+export interface GridRowsInternalCacheState {
+  value: GridRowInternalCacheValue;
+  /**
+   * The value of the properties used by the grouping when the internal cache was created
+   * We are storing it instead of accessing it directly when storing the cache to avoid synchronization issues
+   */
+  props: Pick<DataGridProcessedProps, 'rowCount' | 'getRowId'>;
+  /**
+   * The rows as they were the last time all the rows have been updated at once
+   * It is used to avoid processing several time the same set of rows
+   */
+  rowsBeforePartialUpdates: GridRowsProp;
+}
+
+export interface GridRowsInternalCache {
+  state: GridRowsInternalCacheState;
+  timeout: NodeJS.Timeout | null;
+  lastUpdateMs: number;
```

Comment:
Before this PR, they were not in the state (if by state you mean `GridState`) but in the internal cache of useGridRows which is a ref.
Looks like you added them in the state to be able to access them in `useGridRows` and `rowsStateInitializer`.
Maybe it's a necessary costs of this new implementation.

The more we fragment a feature logic into several hooks and methods, the more this problem will occur.
And if at some point our state becomes a real "react state" (and causes rerenders), we won't be able to store those on the state because they update way to frequently.

Maybe we could introduce an `apiRef.current.caches` collection which would always behave as a ref, and never cause re-renders.

Reply:
I'm was referring to state because it's now in the "state", but thinking it like the old internal cache also works. I didn't understand why these two values were inside an object while they could be standalone refs. I'll open another PR to keep them only inside this hook, not leaking to the state.